From: Kieran Spear <kispear@gmail.com>
Date: Tue, 4 Dec 2012 16:37:34 +1100
Subject: Add support for (cinder) volume quotas

Change-Id: I63b156a6cab3f89295d561d11b596b24d3dbf764

Gbp-Pq-Topic: nectar
Gbp-Pq-Name: volume_quotas.patch
Gbp-Pq-Branch: nectar/volume_quotas
---
 horizon/api/nova.py                                |    7 ++++++-
 .../templates/horizon/common/_quota_summary.html   |    8 ++++++++
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/horizon/api/nova.py b/horizon/api/nova.py
index 3f57a43..08a881d 100644
--- a/horizon/api/nova.py
+++ b/horizon/api/nova.py
@@ -70,6 +70,9 @@ class QuotaSet(object):
     """
     def __init__(self, apiresource):
         self.items = []
+        self.add_quotas(apiresource)
+
+    def add_quotas(self, apiresource):
         for k in apiresource._info.keys():
             if k in ['id']:
                 continue
@@ -405,7 +408,9 @@ def server_remove_floating_ip(request, server, floating_ip):
 
 
 def tenant_quota_get(request, tenant_id):
-    return QuotaSet(novaclient(request).quotas.get(tenant_id))
+    qs = QuotaSet(novaclient(request).quotas.get(tenant_id))
+    qs.add_quotas(cinderclient(request).quotas.get(tenant_id))
+    return qs
 
 
 def tenant_quota_update(request, tenant_id, **kwargs):
diff --git a/horizon/templates/horizon/common/_quota_summary.html b/horizon/templates/horizon/common/_quota_summary.html
index 092bdd3..c0a9ce3 100644
--- a/horizon/templates/horizon/common/_quota_summary.html
+++ b/horizon/templates/horizon/common/_quota_summary.html
@@ -10,4 +10,12 @@
 
     <strong>{% trans "Used" %} <span> {{ usage.quotas.ram.used|intcomma }} MB </span>{% trans "of" %}<span> {{ usage.quotas.ram.quota|intcomma }} MB </span>{% trans "Available RAM" %} </strong>
     {% horizon_progress_bar usage.quotas.ram.used usage.quotas.ram.quota %}
+
+    {% if usage.quotas.volumes %}
+      <strong>{% trans "Used" %} <span> {{ usage.quotas.volumes.used|intcomma }} </span>{% trans "of" %}<span> {{ usage.quotas.volumes.quota|intcomma }} </span>{% trans "Available volumes" %} </strong>
+      {% horizon_progress_bar usage.quotas.volumes.used usage.quotas.volumes.quota %}
+
+      <strong>{% trans "Used" %} <span> {{ usage.quotas.gigabytes.used|intcomma }} GB </span>{% trans "of" %}<span> {{ usage.quotas.gigabytes.quota|intcomma }} GB </span>{% trans "Available volume storage" %} </strong>
+      {% horizon_progress_bar usage.quotas.gigabytes.used usage.quotas.gigabytes.quota %}
+    {% endif %}
 </div>
